graph/graph.go

```diff
@@ -485,10 +516,51 @@ func (graph *Graph) RawJSON(id string) ([]byte, error) {
 }
 
 func jsonPath(root string) string {
-	return filepath.Join(root, "json")
+	return filepath.Join(root, jsonFileName)
 }
 
 // TarLayer returns a tar archive of the image's filesystem layer.
 func (graph *Graph) TarLayer(img *image.Image) (arch archive.Archive, err error) {
-	return graph.driver.Diff(img.ID, img.Parent)
+	root := graph.imageRoot(img.ID)
+	mFileName := filepath.Join(root, tarDataFileName)
+	mf, err := os.Open(mFileName)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			logrus.Errorf("failed to open %q: %s", mFileName, err)
+		}
+		logrus.Debugf("[graph] TarLayer with traditional differ: %s", img.ID)
+		return graph.driver.Diff(img.ID, img.Parent)
+	}
+	pR, pW := io.Pipe()
+	// this will need to be in a goroutine, as we are returning the stream of a
```

Comment:
Is there any way to capture this abstraction in the tar-split library? If metaUnpacker and fileGetter managed their constituent resources, mfz and fsLayer, as `io.ReadCloser`, this would be unnecessary.